core/src/ledger_cleanup_service.rs

```diff
@@ -294,11 +321,11 @@ mod tests {
         // Slot 0 will exist in blockstore with zero shreds since it is slot
         // 1's parent. Thus, slot 0 will be identified for clean.
         assert!(should_clean && lowest_purged == 0);
-        // Now, set max_ledger_shreds to 1 so that slot 0 is left alone
+        // Now, set max_ledger_shreds to 1, slot 0 still eligible for clean
         let max_ledger_shreds = 1;
         let (should_clean, lowest_purged, _) =
             LedgerCleanupService::find_slots_to_clean(&blockstore, last_root, max_ledger_shreds);
-        assert!(!should_clean && lowest_purged == 0);
+        assert!(should_clean && lowest_purged == 0);
```

Comment:
This is kind of an edge case that I don't think is very important. But, the old behavior from this below snippet prevented the root itself from being cleaned (note `>` instead of `>=`):
```rust
            // Unrooted slots are not eligible for cleaning
            if slot > root {
                break;
            }
```
The new behavior allows cleaning the root; I think the new behavior is good and the old comment is inconsistent with the logic.